File: 022.rb
scores = ("A".."Z").to_a

File.open("input/022.txt", "r") do |file|
  names = file.read.split("\",\"")
  names[0]  = names[0][1..-1]
  names[-1] = names[-1][0..-2]
  names = names.sort

  result = 0
  (0...names.size).each do |name|
    score = 0
    names[name].split('').each do |char|
      score += scores.index(char) + 1
    end
    result += (score * (name + 1))
  end

  puts result
end

File: 022.py
def solution022():
    with open("input/022.txt", "r") as file:
        names = [x[1:-1] for x in file.read().split(",")]

        names = sorted(names)

        result = 0
        for name in range(0, len(names)):
            score = 0
            for c in names[name]:
                score += ord(c) - ord("A") + 1
            result += score * (name + 1)

    return result
